velox/functions/sparksql/tests/StringTest.cpp

```diff
@@ -418,5 +442,34 @@ TEST_F(StringTest, substring) {
   EXPECT_EQ(substring("da\u6570\u636Eta", -3), "\u636Eta");
 }
 
+TEST_F(StringTest, overlay) {
+  EXPECT_EQ(overlay("Spark\u6570\u636ESQL", "_", 6, -1), "Spark_\u636ESQL");
+  EXPECT_EQ(overlay("Spark SQL", "_", 6, -1), "Spark_SQL");
+  EXPECT_EQ(overlay("Spark SQL", "CORE", 7, -1), "Spark CORE");
+  EXPECT_EQ(overlay("Spark SQL", "ANSI ", 7, 0), "Spark ANSI SQL");
+  EXPECT_EQ(overlay("Spark SQL", "tructured", 2, 4), "Structured SQL");
+
+  EXPECT_EQ(overlay("Spark SQL", "##", 10, -1), "Spark SQL##");
+  EXPECT_EQ(overlay("Spark SQL", "##", 10, 4), "Spark SQL##");
+  EXPECT_EQ(overlay("Spark SQL", "##", 0, -1), "##park SQL");
+  EXPECT_EQ(overlay("Spark SQL", "##", 0, 4), "##rk SQL");
+  EXPECT_EQ(overlay("Spark SQL", "##", -10, -1), "##park SQL");
+  EXPECT_EQ(overlay("Spark SQL", "##", -10, 4), "##rk SQL");
+}
+
+TEST_F(StringTest, overlayVarbinary) {
```

Comment:
What's the difference in varchar and varbinary implementations? Is it that in varchar 'len' is in characters and in varbinary 'len' is in bytes?

Reply:
After debug Gluten's code, if Spark side's input is `overlay(varbinary(1, 2, 3, 4, 5, 6, 7, 8, 9), varbinary(-1, -1, -1, -1, -1), 2, 4)`, Velox side's plan is 
```
-- Project[expressions: (n1_1:VARBINARY, overlay("AQIDBAUGBwgJ","//////8=",2,4))] -> n1_1:VARBINARY
  -- ValueStream[] -> n0_0:INTEGER
```

> Is it that in varchar 'len' is in characters and in varbinary 'len' is in bytes?

I think `len` is in characters.